glean-core/python/glean/_dispatcher.py

```diff
@@ -18,22 +135,36 @@ class Dispatcher:
     # are run immediately
     _queue_initial_tasks = True  # type: bool
 
-    # The task queue
-    _task_queue = []  # type: List[Tuple[Callable, tuple, dict]]
+    # The preinit task queue
+    _preinit_task_queue = []  # type: List[Tuple[Callable, tuple, dict]]
+
+    # The live task queue to run things in another thread
+    _task_worker = _ThreadWorker()
 
     # The number of tasks that overflowed the queue
     _overflow_count = 0  # type: int
 
+    # When `True`, all tasks are run synchronously
+    _testing_mode = False  # type: bool
+
     @classmethod
     def reset(cls):
         """
         Reset the dispatcher so the queue is cleared, and it is reset into
         queueing mode.
         """
         cls._queue_initial_tasks = True
-        cls._task_queue = []
+        cls._preinit_task_queue = []
         cls._overflow_count = 0
 
+    @classmethod
+    def _execute_task(cls, func: Callable, *args, **kwargs):
```

Comment:
Since we're now supporting priorities, we should also make sure to add test coverage for [this case](https://searchfox.org/glean/rev/55bba00886c39f2291906467ea519c6457f3a552/glean-core/android/src/test/java/mozilla/telemetry/glean/GleanTest.kt#447).

It would also be relevant to test (I swear there's a test for that in Kotlin, but can't find it!) that 'application' lifetime metrics are cleared *after* the Glean-owned pings are generated.